shell/platform/darwin/ios/framework/Source/SemanticsObject+UIFocusSystem.mm

```diff
@@ -71,8 +81,56 @@ - (BOOL)canBecomeFocused {
   return self.node.HasAction(flutter::SemanticsAction::kTap);
 }
 
+// The frame is described in the `coordinateSpace` of the
+// `parentFocusEnvironment` (all `parentFocusEnvironment`s are `UIFocusItem`s).
+//
+// Seealso the `coordinateSpace` implementation.
 - (CGRect)frame {
-  return self.accessibilityFrame;
+  SkPoint quad[4] = {SkPoint::Make(self.node.rect.left(), self.node.rect.top()),
+                     SkPoint::Make(self.node.rect.left(), self.node.rect.bottom()),
+                     SkPoint::Make(self.node.rect.right(), self.node.rect.top()),
+                     SkPoint::Make(self.node.rect.right(), self.node.rect.bottom())};
+
+  SkM44 transform = self.node.transform;
+  FlutterSemanticsScrollView* scrollView;
+  for (SemanticsObject* ancestor = self.parent; ancestor; ancestor = ancestor.parent) {
+    if ([ancestor isKindOfClass:[FlutterScrollableSemanticsObject class]]) {
+      scrollView = ((FlutterScrollableSemanticsObject*)ancestor).scrollView;
+      break;
+    }
+    transform = ancestor.node.transform * transform;
+  }
+
+  for (auto& vertex : quad) {
+    SkV4 vector = transform.map(vertex.x(), vertex.y(), 0, 1);
+    vertex = SkPoint::Make(vector.x / vector.w, vector.y / vector.w);
+  }
+
+  SkRect rect;
+  rect.setBounds(quad, 4);
```

Comment:
I'm presuming the reason for using Skia types in the first half of the methods is that `self.node.transform` was already a SkM44? That seems reasonable given the existing code but would you mind adding a TODO to migrate this code to CoreGraphics types? The less Skia-dependent code we can add in the embedders themselves, the better.

Reply:
I basically copied this from https://github.com/flutter/engine/blob/764b91d7d75af08a334a1509cd24cdefcc16f2e5/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L66-L95. 

But here `setBounds` is probably just taking the `MIN` and `MAX` of the coordinates. I'll do that by hand.